src/resolve/local.ts

```diff
@@ -17,61 +13,35 @@ export default async function resolveLocal (spec: PackageSpec, opts: ResolveOpti
 
   if (dependencyPath.slice(-4) === '.tgz' || dependencyPath.slice(-7) === '.tar.gz') {
     const name = getTarballName(dependencyPath)
-    return {
+    const resolution: Resolution = {
+      type: 'tarball',
       id: createLocalPkgId(name, dependencyPath),
-      root: path.dirname(dependencyPath),
-      fetch: createLocalTarballFetcher({
-        tarball: dependencyPath
-      })
+      tarball: `file:${dependencyPath}`,
     }
+    return {resolution}
   }
 
   if (opts.linkLocal) {
```

Comment:
Shrinkwrap might not be the right place for this, at least in the form it is now — mapping from version requests to exact and stable identifiers. Because if we have two A packages resolved to the same version but only want to set `link-local`/`no-optional`/`no-bin-links` to the one of them then we won't be able to store such info in shrinkwrap because it sees them as the same record.

Actually I thought of that before — what if shrinkwrap is not a mapping from versions requests to ids but a tree which describes/mirrors the layout of packages in a filesystems? That way we could attach options to any of the package in the dep graph. This is actually super useful for Esy as Esy would need to store info such CPU architecture for cross compilation support.